### ISSUE TITLE
ByteArrayOutputStream should be an AutoCloseable

### ISSUE THREAD

**Issue by Atry** [scala-js/scala-js#3107]
``` scala
val a: AutoCloseable = new java.io.ByteArrayOutputStream
a.close()
```

```
ERROR: this.a$1.close__V is not a function
```

https://scalafiddle.io/sf/SOUCOqB/0

**Comment by xuwei-k** [comment]
https://github.com/scala-js/scala-js/blob/v0.6.19/javalib/src/main/scala/java/io/Closeable.scala

> Note that Closeable doesn't extend AutoCloseable for Java6 compat

**Comment by Atry** [comment]
Related discussion on Gitter: https://gitter.im/scala-js/scala-js/core?at=599c4ca92723db8d5ecf5689

**Comment by sjrd** [comment]
Fixed in 670144bf31b80a33a36f98a50a85c30bb4d83012.
